src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1025,19 +1055,86 @@ private MethodSignatureFlags PlatformDefaultUnmanagedCallingConvention()
                 MethodSignatureFlags.UnmanagedCallingConventionStdCall : MethodSignatureFlags.UnmanagedCallingConventionCdecl;
         }
 
-        private CorInfoUnmanagedCallConv getUnmanagedCallConv(CORINFO_METHOD_STRUCT_* method)
+        private CorInfoCallConvExtension getUnmanagedCallConv(CORINFO_METHOD_STRUCT_* method, CORINFO_SIG_INFO* sig, ref bool pSuppressGCTransition)
         {
-            MethodSignatureFlags unmanagedCallConv = HandleToObject(method).GetPInvokeMethodMetadata().Flags.UnmanagedCallingConvention;
+            pSuppressGCTransition = false;
+
+            if (method != null)
+            {
+                MethodDesc methodDesc = HandleToObject(method);
+                CorInfoCallConvExtension callConv = GetUnmanagedCallConv(HandleToObject(method), out pSuppressGCTransition);
+                return callConv;
+            }
+            else
+            {
+                Debug.Assert(sig != null);
 
-            if (unmanagedCallConv == MethodSignatureFlags.None)
-                unmanagedCallConv = PlatformDefaultUnmanagedCallingConvention();
+                CorInfoCallConvExtension callConv = GetUnmanagedCallConv((MethodSignature)HandleToObject((IntPtr)sig->pSig), out pSuppressGCTransition);
+                if (!pSuppressGCTransition && sig->flags.HasFlag(CorInfoSigInfoFlags.CORINFO_SIGFLAG_SUPPRESS_GC_TRANSITION))
```

Comment:
```suggestion
                if (sig->flags.HasFlag(CorInfoSigInfoFlags.CORINFO_SIGFLAG_SUPPRESS_GC_TRANSITION))
```
`!pSuppressGCTransition` will be true 98+% of the time.